File: 1475.商品折扣后的最终价格.py
#
# @lc app=leetcode.cn id=1475 lang=python3
#
# [1475] 商品折扣后的最终价格
#

# @lc code=start
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        stack, result = [], [i for i in prices]
        for i in range(len(prices)):
            while stack and prices[stack[-1]] >= prices[i]:
                result[stack[-1]] -= prices[i]
                stack.pop()
            stack.append(i)
        return result
# @lc code=end

File: 1475.商品折扣后的最终价格.cpp
/*
 * @lc app=leetcode.cn id=1475 lang=cpp
 *
 * [1475] 商品折扣后的最终价格
 */

// @lc code=start
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        for(int i = 0; i < prices.size(); i++){
            for(int j = i+1; j < prices.size(); j++){
                if(prices[j]<= prices[i])
                {
                    prices[i] -= prices[j];
                    break;
                }
            }
        }
        return prices;
    }
};
// @lc code=end

